source/world.cc

```diff
@@ -151,6 +152,21 @@ namespace WorldBuilder
     prm.coordinate_system = prm.get_unique_pointer<CoordinateSystems::Interface>("coordinate system");
     prm.coordinate_system->parse_entries(prm);
 
+
+    // For now allow a value of ININITY to escape the check. This escape will
+    // be removed in future versions.
+    double max_model_depth = prm.coordinate_system->max_model_depth();
+    WBAssertThrow(max_model_depth_external == INFINITY
+                  ||
+                  std::fabs(max_model_depth_external-max_model_depth)
```

Comment:
As I mentioned in #331, this assertion assumes that the external code has an opinion on the geometry of the problem. This may be the case, as in the case of ASPECT, or it may not: External codes might simply consider WorldBuilder's geometry as the ground truth and not build a separate representation. I would move this assertion to ASPECT (in other words: remove it here, and when this patch is merged, add it to ASPECT).